plugins/auth-backend/src/providers/azure/provider.ts

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import express from 'express';
+
+import { OIDCStrategy as AzureStrategy } from 'passport-azure-ad';
+
+import {
+  OAuthProviderHandlers,
+  OAuthResponse,
+  PassportDoneCallback,
+  RedirectInfo,
+  AuthProviderConfig,
+  AzureProviderOptions,
+} from '../types';
+
+import {
+  executeRedirectStrategy,
+  executeFrameHandlerStrategy,
+  executeRefreshTokenStrategy,
+  executeFetchUserProfileStrategy,
+  makeProfileInfo,
+} from '../../lib/PassportStrategyHelper';
+
+import { OAuthProvider } from '../../lib/OAuthProvider';
+import { Logger } from 'winston';
+import { TokenIssuer } from '../../identity';
+import { Config } from '@backstage/config';
+import passport from 'passport';
+
+import got from 'got';
```

Comment:
Used to fetch the user's photo, which is not returned in the token and must be retrieved from the Microsoft Graph API's `/me` endpoint.